src/panels/lovelace/cards/hui-gauge-card.ts

```diff
@@ -174,6 +174,26 @@ class HuiGaugeCard extends LitElement implements LovelaceCard {
     if (this._config!.needle) {
       return undefined;
     }
+
+    // new format
+    let segments = this._config!.segments;
+    if (segments) {
+      segments = [...segments].sort((a, b) => a?.from - b?.from);
+
+      for (let i = 0; i < segments.length; i++) {
+        const segment = segments[i];
+        if (
+          segment &&
+          numberValue >= segment.from &&
+          (i + 1 === segments.length || numberValue < segments[i + 1]?.from)
+        ) {
+          return segment.color;
```

Comment:
```suggestion
        if (
          numberValue >= segment.from &&
          (i + 1 === segments.length || numberValue < segments[i + 1].from)
        ) {
          return segment.color;
```

Reply:
This change and the others you commented will lead to errors while editing the card config. IMHO it's nicer to catch theses cases to prevent spamming the console.
![image](https://user-images.githubusercontent.com/16437929/164720083-82630f2e-294d-43c6-a56f-26b813a94548.png)

Reply:
ah yes. That makes sense 👍